curve_apps/edge_detection/params.py

```diff
@@ -72,11 +83,19 @@ class Parameters(BaseData):
 
     _name: str = NAME
 
-    input_file: InputFile | None = InputFile.read_ui_json(
-        DEFAULT_UI_JSON, validate=False
-    )
+    input_file: InputFile = InputFile.read_ui_json(DEFAULT_UI_JSON, validate=False)
     detection: DetectionParameters
     output: OutputParameters
     run_command: str = f"curve_apps.{NAME}.driver"
     source: SourceParameters
     title: str = NAME.capitalize().replace("_", " ")
+
+    @model_validator(mode="after")
+    def update_input_file(self):
+        if self.input_file is None or not self.input_file.validate:
+            params_data = self.flatten()
+            if self.input_file.ui_json is None:
+                raise ValueError("Input file should have a ui_json set.")
+            data = flatten(self.input_file.ui_json)
+            data.update(params_data)
+            self.input_file = InputFile(data=data, ui_json=self.input_file.ui_json)
```

Comment:
Why not just updating data with current values?

Reply:
I don't remember.. it's what worked at the time?

Reply:
Maybe to retain the validation to match what the og params classes were doing?

Reply:
yep, able to clean it out now since the params classes are gone now